control/veda_val.py

```diff
@@ -356,9 +356,11 @@ def send_200(self, val_api_return):
         VAL will not allow duped studio urls to be sent,
         so we must scrub the data
         """
-        for c in val_api_return['courses']:
-            if c in self.val_data['courses']:
-                self.val_data['courses'].remove(c)
+        for course in val_api_return['courses']:
+            for course_id in course.keys():
+                for course_entry in self.val_data['courses']:
```

Comment:
This is not good, we shouldn't iterate and remove on the same list, it would give unexpected results.